services/director-v2/src/simcore_service_director_v2/cli.py

```diff
@@ -0,0 +1,131 @@
+import asyncio
+import logging
+from contextlib import asynccontextmanager
+from typing import AsyncIterator, Final
+from uuid import UUID
+
+import httpx
+import typer
+from fastapi import FastAPI
+from models_library.projects import NodeIDStr, ProjectID
+from pydantic import AnyHttpUrl, parse_obj_as
+from servicelib.fastapi.long_running_tasks.client import ClientConfiguration
+from settings_library.utils_cli import create_settings_command
+from tenacity._asyncio import AsyncRetrying
+from tenacity.stop import stop_after_attempt
+from tenacity.wait import wait_random_exponential
+
+from .core.settings import AppSettings
+from .meta import PROJECT_NAME
+from .models.schemas.dynamic_services import DynamicSidecarNames
+from .modules import db
+from .modules.db.repositories.projects import ProjectsRepository
+from .modules.director_v0 import DirectorV0Client
+from .modules.dynamic_sidecar import api_client
+from .modules.dynamic_sidecar.scheduler.events_utils import (
+    fetch_repo_outside_of_request,
+)
+from .modules.projects_networks import requires_dynamic_sidecar
+
+DEFAULT_NODE_SAVE_RETRY: Final[int] = 3
+
+main = typer.Typer(name=PROJECT_NAME)
+
+log = logging.getLogger(__name__)
+main.command()(create_settings_command(settings_cls=AppSettings, logger=log))
+
+
+@asynccontextmanager
+async def _initialized_app(settings: AppSettings) -> AsyncIterator[FastAPI]:
```

Comment:
Ideally use this as part of the main app initialization function. This way both are in sync.

For instance, when the service logic evolves (e.g. new) this part is consistent with it

Reply:
look at d``create_base_app`` design of dynamic-sidecar. It creates a lightweight base application

Reply:
I agree. It would be much easier to maintain if we use common code rather than a separate branch of code.